src/app/update-password/update-password.page.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, OnInit } from '@angular/core';
+import {
+  FormBuilder,
+  FormGroup,
+  Validators,
+} from '@angular/forms';
+import { Platform, ToastController } from '@ionic/angular';
+import { Storage } from '@ionic/storage';
+import { FirebaseService } from '../services/firebase.service';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-update-password',
+  templateUrl: './update-password.page.html',
+  styleUrls: ['./update-password.page.scss'],
+})
+export class UpdatePasswordPage implements OnInit {
+
+  public updatePasswordForm: FormGroup;
+  passwordType = 'password';
+  passwordText = 'SHOW';
+  newPasswordType = 'password'
+  newPasswordText = 'SHOW'
+  isIOS = false;
+  userData:any;
+
+  constructor(private readonly formBuilder: FormBuilder,
+    private readonly router: Router,
+    public platform: Platform,
+    private firebaseService: FirebaseService,
+    private toastController: ToastController,
+    private storage: Storage,) { }
+
+
+  ngOnInit() {
+    this.initializeUpdatePasswordForm();
+    this.storage.get('userData').then((val) => {
+      this.userData = val;
+      console.log('userData', this.userData);
+    });
+  }
+
+  initializeUpdatePasswordForm() {
+    this.updatePasswordForm = this.formBuilder.group({
+      oldPassword: ["",
+        Validators.compose([
+          Validators.minLength(8),
+          Validators.required
+        ])
+      ],
+
+      newPassword: [
+        "",
+        Validators.compose([
+          Validators.minLength(8),
+          Validators.required
+        ])
+      ],
+    });
+    this.isIOS = this.platform.is('ios');
+  }
+
+  hideShowPassword(): void {
+    this.passwordType = this.passwordType === 'text' ? 'password' : 'text';
+    this.passwordText = this.passwordText === 'SHOW' ? 'HIDE' : 'SHOW';
+  }
+
+  hideShowNewPassword(): void {
+    this.newPasswordType = this.newPasswordType === 'text' ? 'password' : 'text';
+    this.newPasswordText = this.newPasswordText === 'SHOW' ? 'HIDE' : 'SHOW';
+  }
+
+  updatePassword() {
+    const oldPassword = this.updatePasswordForm.get('oldPassword').value
+    const newPassword = this.updatePasswordForm.get('newPassword').value
+    if(oldPassword === this.userData.password){
```

Comment:
Why we need to check old pass and new pass?

Reply:
@dtrmangesh to check the entered password is correct or not and on then updating new password.